### ISSUE TITLE
False-positive references

### ISSUE THREAD

**Issue by jameslkingsley** [DEVSENSE/phptools-docs#580]
If I lookup the references to a function, PHP tools will show me false-positives in cases where no types could be determined.

Example of this is where you have mocks in a test that are testing a completely unrelated class:

So I have a trait such as:

```php
trait UsesAttributes
{
    public function getAttribute(string $type)
    {
        ...
    }
}
```

And then somewhere completely unrelated in a test:

```php
$thing = Mockery::mock(Thing::class);
$thing->expects()->getAttribute('id')->atLeast()->once()->andReturns(1);
```

Even though it's using a method named `getAttribute`, it has nothing to do with the trait, it doesn't use it at all. Yet PHP tools tries to be clever and assumes this test is making use of it.

There needs to be a setting to only consider references to things when there is absolute confidence in the types being used.

**Comment by jakubmisek** [comment]
Thank you for pointing this out.

Internally, we know how confident the reference is (it's used during rename refactoring for example). When finding all the references, we wanted to highlight all the potential ones, so we don't miss any (most PHP code is not annotated well so users would miss a lot of references). Sadly there is no way in VSCode to distinguish between potential and solid reference.

We can add a setting, or maybe change the default behavior.

**Comment by jameslkingsley** [comment]
Yeah a setting would be ideal I think. Depends on the maturity of the codebase, which in my case I'm happy for that tradeoff of potentially missing some older code
